Vercel edge middleware targets es2020, which doesn't support top level await

### Astro Info

```block
Astro                    v5.1.2
Node                     v22.12.0
System                   Windows (x64)
Package Manager          npm
Output                   server
Adapter                  @astrojs/vercel
Integrations             @astrojs/tailwind
                         @astrojs/sitemap
                         astro-robots-txt
                         @astrojs/partytown
                         @astrojs/web-vitals
                         astro:db
                         @astrojs/db/file-url
```


### If this issue only occurs in one browser, which browser is a problem?

Chrome, Brave

### Describe the Bug

When I upgraded my project to the latest version of Astro, astro:db started giving errors. The error is available below;
```
11:13:35 [build] Rearranging server assets...
11:13:35 [@astrojs/vercel] Bundling function ..\..\..\..\dist\server\entry.mjs
✘ [ERROR] Top-level await is not available in the configured target environment ("es2020")

    dist/server/chunks/_astro_db_HwI--WRx.mjs:3:11:
      3 │ const db = await createRemoteDatabaseClient({
        ╵            ~~~~~

11:13:38 [build] Waiting for integration "@astrojs/vercel", hook "astro:build:done"...
Top-level await is not available in the configured target environment ("es2020")
  Location:
    dist/server/chunks/_astro_db_HwI--WRx.mjs:3:11
  Stack trace:
    at failureErrorWithLog (C:\Users\Administrator\Desktop\...\node_modules\esbuild\lib\main.js:1476:15)
    at runOnEndCallbacks (C:\Users\Administrator\Desktop\...\node_modules\esbuild\lib\main.js:1316:45)
    at C:\Users\Administrator\Desktop\...\node_modules\esbuild\lib\main.js:970:16
    at handleIncomingPacket (C:\Users\Administrator\Desktop\...\node_modules\esbuild\lib\main.js:677:12)
    at Socket.emit (node:events:524:28)
```

I am try to fix astro config;

```
vite: {
    server: {
      host: true,
      open: true
    },
    build: {
      minify: 'esbuild',
      target: 'es2022'
    }
  }
```

and with tsconfig.json file;

```
{
  "extends": "astro/tsconfigs/strict",
  "compilerOptions": {
    "strictNullChecks": true,
    "allowImportingTsExtensions": true,
    "baseUrl": ".",
    "target": "ESNext",
    "paths": {
      "@/*": ["src/*"]
    }
  }
}
```

But it doesnt working.

### What's the expected result?

I am waiting for the project to be built properly and published on the `Vercel` server.

### Link to Minimal Reproducible Example

https://astro.new/repro

### Participation

- [ ] I am willing to submit a pull request for this issue.

--- a/.changeset/grumpy-phones-explode.md
+++ b/.changeset/grumpy-phones-explode.md
@@ -1,0 +1,5 @@
+---
+'@astrojs/vercel': patch
+---
+
+Updates edge middleware to support es2022 syntax

--- a/.changeset/grumpy-phones-explode.md
+++ b/.changeset/grumpy-phones-explode.md
@@ -2,4 +2,4 @@
 '@astrojs/vercel': patch
 ---
 
-Updates edge middleware to support es2022 syntax
+Updates edge middleware to support esnext syntax

--- a/packages/vercel/src/serverless/middleware.ts
+++ b/packages/vercel/src/serverless/middleware.ts
@@ -42,10 +42,12 @@ export async function generateEdgeMiddleware(
 			contents: code,
 			resolveDir: fileURLToPath(root),
 		},
-		target: 'es2020',
+		target: 'es2022',
 		platform: 'browser',
+		// esbuild automatically adds the browser, import and default conditions
+		// https://esbuild.github.io/api/#conditions
 		// https://runtime-keys.proposal.wintercg.org/#edge-light
-		conditions: ['edge-light', 'worker', 'browser'],
+		conditions: ['edge-light', 'workerd', 'worker'],
 		outfile: bundledFilePath,
 		allowOverwrite: true,
 		format: 'esm',

--- a/packages/vercel/src/serverless/middleware.ts
+++ b/packages/vercel/src/serverless/middleware.ts
@@ -42,7 +42,9 @@ export async function generateEdgeMiddleware(
 			contents: code,
 			resolveDir: fileURLToPath(root),
 		},
-		target: 'es2022',
+		// Vercel Edge runtime targets ESNext, because Cloudflare Workers update v8 weekly
+		// https://github.com/vercel/vercel/blob/1006f2ae9d67ea4b3cbb1073e79d14d063d42436/packages/next/scripts/build-edge-function-template.js
+		target: 'esnext',
 		platform: 'browser',
 		// esbuild automatically adds the browser, import and default conditions
 		// https://esbuild.github.io/api/#conditions

--- a/packages/vercel/test/fixtures/middleware-with-edge-file/src/middleware.js
+++ b/packages/vercel/test/fixtures/middleware-with-edge-file/src/middleware.js
@@ -1,3 +1,7 @@
+export const hello = async () => 'hello world';
+
+const message = await hello();
+
 /**
  * @type {import("astro").MiddlewareResponseHandler}
  */
@@ -5,5 +9,6 @@ export const onRequest = async (context, next) => {
 	const test = 'something';
 	context.cookies.set('foo', 'bar');
 	const response = await next();
+	response.headers.set('x-message', message);	
 	return response;
 };

